osbrain/tests/test_agent_sync_publications.py

```diff
@@ -283,3 +283,41 @@ def test_wait(nsproxy):
     assert wait_agent_attr(client, name='error_log', length=1, timeout=0.5)
     assert server.get_attr('received') == [fast, slow]
     assert 'x' + str(slow) not in client.get_attr('received')
+
+
+def test_various_requests(nsproxy):
```

Comment:
I'm not sure this test belongs here. Maybe we are just testing:

- ZeroMQ behavior (i.e.: the poller which processes messages from multiple sockets in a round-robin fashion), which should be tested by `pyzmq` itself
- osBrain general behavior (i.e.: all input messages, by default, are processed sequentially by the main thread), which should be tested elsewhere and is probably tested already.

Once it is done, though, it makes no harm (and maybe it actually tests something related to sync. PUB-SUB that we missed before). :blush: